lib/ansiblelint/rules/NestedJinjaRule.py

```diff
@@ -0,0 +1,45 @@
+# Copyright (c) 2020, Red Hat, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+import re
+from ansiblelint import AnsibleLintRule
+
+
+class NestedJinjaRule(AnsibleLintRule):
+    id = '207'
+    shortdesc = 'Nested jinja pattern'
+    description = (
+        'There should not be any nested jinja pattern. '
+        'Example (bad): \'{{ list_one + {{ list_two | max }} }}\', '
+        'example (good): \'{{ list_one + max(list_two) }}\''
+    )
+    severity = 'VERY_HIGH'
+    tags = ['formatting']
+    version_added = 'v4.2.1'
+
+    nested_jinja_pattern = re.compile(r"{{(?:[^{}]*)?{{")
```

Comment:
Does this really need to exist outside the `matchtask` method? If yes, I'd put it on the module level as `NESTED_JINJA_PATTERN = re.compile(r"{{(?:[^{}]*)?{{")` rather than in a class.

Reply:
I was inspired by other rules where it was done the same.

E.g.: https://github.com/ansible/ansible-lint/blob/master/lib/ansiblelint/rules/VariableHasSpacesRule.py#L16

It can be found in the following files too:
* `ComparisonToEmptyStringRule.py`
* `ComparisonToLiteralBoolRule.py`
* `MetaTagValidRule.py`
* `MetaVideoLinksRule.py`
* `MetaVideoLinksRule.py`
* `MetaVideoLinksRule.py`
* `NestedJinjaRule.py`
* `ShellWithoutPipefail.py`
* `ShellWithoutPipefail.py`
* `UsingBareVariablesIsDeprecatedRule.py`
* `UsingBareVariablesIsDeprecatedRule.py`
* `VariableHasSpacesRule.py`
* `VariableHasSpacesRule.py`

Do you want to move it there anyway?

Reply:
Ah, nevermind then. A Foolish Consistency is the Hobgoblin of Little Minds.